eip712-adapter-moshi/src/main/java/pm/gnosis/eip712/adapters/moshi/MoshiAdapter.kt

```diff
@@ -0,0 +1,61 @@
+package pm.gnosis.eip712.adapters.moshi
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import okio.buffer
+import okio.source
+import pm.gnosis.eip712.JsonAdapter
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.nullOnThrow
+import java.io.InputStream
+
+class MoshiAdapter : JsonAdapter {
+    private val moshi by lazy { Moshi.Builder().build() }
+
+    override fun parse(rawJson: String): JsonAdapter.Result {
```

Comment:
Would be nice what `rawJson: String` is expected to be here.

For me it is not clear/ doesn'T make sense, why this is the json rpc payload.

Reply:
What's your suggestion here?

Reply:
For me it should be just the json of the typed data for which we want to create the hash. Also adding a comment that clearifies this might be good (in the parent class and the EIP712 class)

Reply:
Fixed in https://github.com/gnosis/svalinn-kotlin/pull/31/commits/10cc16042762e85f3bc4b6a25f4241cec57d8a94